main.tf

```diff
@@ -192,10 +192,11 @@ resource "aws_autoscaling_group" "agent_asg" {
   max_size = var.agent_max
   min_size = var.agent_min
 
+  # added cooling period for ASG to pick up another scaling activity after one is complete
   health_check_grace_period = 300
   health_check_type         = "EC2"
-
-  name = "${var.application}-agent-asg"
+  name                      = "${var.application}-agent-asg"
+  default_cooldown          = 100
```

Comment:
@ashish-akshantal how did you come to chose 100 seconds as the appropriate amount of time before another event takes place? The default value is 500 seconds, why reduce this value? https://docs.aws.amazon.com/autoscaling/ec2/userguide/ec2-auto-scaling-scaling-cooldowns.html